server/embed/config.go

```diff
@@ -877,6 +908,25 @@ func (cfg *configYAML) configFromFile(path string) error {
 	return cfg.Validate()
 }
 
+// SetFeatureGatesFromExperimentalFlags sets the feature gate values if their corresponding experimental flags are
+// explicitly set.
+func (cfg *Config) SetFeatureGatesFromExperimentalFlags(isExperimentalFlagSet func(string) bool, featureGatesVal string) error {
+	// verify that the feature gate and its experimental flag are not both set at the same time.
+	for expFlagName, featureName := range features.ExperimentalFlagToFeatureMap {
+		if isExperimentalFlagSet(expFlagName) && strings.Contains(featureGatesVal, string(featureName)) {
+			return fmt.Errorf("cannot specify both flags: --%s=(true|false) and --%s=%s=(true|false) at the same time, please just use --%s=%s=(true|false)",
+				expFlagName, featuregate.FlagName, featureName, featuregate.FlagName, featureName)
+		}
+	}
+
+	m := make(map[string]bool)
+	defaultEc := NewConfig()
+	if cfg.ExperimentalStopGRPCServiceOnDefrag != defaultEc.ExperimentalStopGRPCServiceOnDefrag {
```

Comment:
Question here - For a function that sounds generic we are only internally only focusing on `ExperimentalStopGRPCServiceOnDefrag` can you please reply with some some context on the intention for this function now and longer term to help me understand?

Reply:
Added some comments in the function. Does that make sense? 
`ExperimentalStopGRPCServiceOnDefrag` is just what we are migrating in this PR. For any experimental flag we are migrating in the future, this function needs to be updated.

Reply:
Thanks for adding the explanation, sgtm.